File: 1TP1/4_D.cpp
#include<iostream>
using namespace std;

int main() {
  int min = 1000000, max = -1000000, n, a;
  long long sum = 0; // 10^6 が 10^4個入力されるため、10^10を保持できる必要がある

  cin >> n;

  for (int i = 0; i < n; i++) {
    cin >> a;
    if (min > a) min = a;
    if (max < a) max = a;
    sum += a;
  }

  cout << min << ' ' << max << ' ' << sum << endl;

  return 0;
}

File: PHP/1TP1/4_D.php
<?php

$n = intval(trim(fgets(STDIN))); // 使わないか
$intArray = array_map('intval', explode(' ', trim(fgets(STDIN))));

// MEMO: 10^10は扱える
// echo PHP_INT_MAX;

$min = 1000000;
$max = -1000000;
$sum = 0; // array_sumで出せる

array_map(
    function ($val) use (&$min, &$max, &$sum) {
        if ($min > $val) $min = $val;
        if ($max < $val) $max = $val;
        $sum += $val;
    },
    $intArray
);

echo "{$min} {$max} {$sum}\n";
